insights: incomplete data troubleshoot insights docs (#45465)
sourcegraph/sourcegraph: doc/code_insights/references/incomplete_data_points.md

To address this, try to minimize or avoid cases where your insight data series: 

1. Runs over extra large sets of repositories (scope your insight further to fewer repositories)
1. Uses many boolean combinations (consider splitting into multiple data series)
1. Runs a complex search over an especially large monorepo (consider optimizing your search query to be more specific or include more filters, like `lang:` or `file:`)

In addition: 

1. Timeout errors on points that have been backfilled before the creation date of the insight are more likely to occur on single, large repositories, because backfill points are calculated by running many searches, repository by repository, individually. 
1. Timeout errors on points that have been snapshot after the creation date of the insight are more likely to occur on insights running complex searches over large numbers of repositories, because snapshot points are calculated by running a single global search against the current index.
You can read more about this case in our [limitations](../explanations/current_limitations_of_code_insights.md#accuracy-considerations-for-an-insight-query-returning-a-large-result-set).

> 
> If the data is available, the error alert should inform you for which times the search has timed out.

Author: ```suggestion
If the data is available, the error alert will inform you which times the search has timed out.
```
